src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -215,7 +215,12 @@ uct_rc_mlx5_iface_handle_failure(uct_ib_iface_t *ib_iface, void *arg,
     ucs_log_level_t    log_lvl;
     ucs_status_t       status;
 
-    ucs_assert(ep != NULL);
+    if (ep == NULL) { /* EP is already destroyed */
```

Comment:
does it mean that we do pointer arithmetic above for RC EP which `NULL`?

Reply:
there is no pointer arithmetic with EP above - just searching EP in table

Reply:
but we do `ucs_derived_of()` after `uct_rc_iface_lookup_ep()`

Reply:
this is just casting, not real processing

Reply:
yes, you're right
we expect `offsetof == 0` for `super` filed, so the result of `uct_rc_iface_lookup_ep()` == `ucs_derived_of()` of this result

Reply:
right, and moreover offsetof doesn't produce access to memory too - it is just arthmetic operation, safe for any pointer

Reply:
i see